src/debugger/handler_dbgp.c

```diff
@@ -2322,6 +2323,19 @@ static int xdebug_dbgp_cmdloop(xdebug_con *context, int bail)
 		free(option);
 	} while (0 == ret);
 
+	if (!XG_DBG(context).do_connect_to_client &&
```

Comment:
When we exit the command loop, if we are only debugging and there are no conditions that require us to use the debugging functions, like a step condition or some breakpoints, then we disable the debugger.

And if there are any of these conditions and the debugger was disabled, we re-enable it